docs/blog/2024-12-28-release-v3.0.0/index.md

```diff
@@ -0,0 +1,91 @@
+---
+title: Woodpecker 3.0.0
+description: Introducing Woodpecker 2.5.0
+slug: release-v300
+authors:
+  - name: pat-s
+    title: Maintainer of Woodpecker
+    url: https://github.com/pat-s
+    image_url: https://github.com/pat-s.png
+tags: [release, major]
+hide_table_of_contents: false
+---
+
+> Disclaimer upfront: The Woodpecker team is aware that this release contains _a lot_ of changes, also many which force users to update their pipeline definitions.
+> We understand that this can be a tedious task, especially when managing numerous repositories and pipelines. Each change was carefully considered and thoroughly discussed, with specific reasoning behind every decision.
+> A significant portion of these updates is focused on “breaking free” from outdated and suboptimal Drone definitions.
+> Thank you for your patience and understanding as we implement these essential breaking changes!
+
+Security has been the major focus in this major release.
+Besides patching known vulnerabilities (and also backporting these to v2 releases), the secrets handling mechanism has been improved, preventing accidental leaks and making it simpler to keep sensitive information fully encrypted.
+
+Specifically, the `secrets:` keyword has been deprecated in favor of a more flexible (and secure) way to specify secrets: `from_secret:`.
+This new approach provides more flexibility (by being to use different names for the source and destination secrets) and ensures a safe internal secret parsing through a unified engine.
+Because secrets defined via `secrets:` were simple env vars in the end, this change also removes potential confusion about the differences between values specified in `environment:` and `secrets`.
+Now, both are defined in `environment:` using an expressive syntax:
+
+```yaml
+steps:
+  name:
+    image: alpine
+    commands:
+      - echo "The secret is $TOKEN_ENV"
+    environment:
+      TOKEN_ENV:
+        from_secret: SECRET_TOKEN
+```
+
+## Rootless images
+
+Woodpecker now supports running rootless images by adjusting the entrypoints and directory permissions in the containers in a way that allows non-privileged users to execute tasks.
+
+In addition, all images published by Woodpecker (Server, Agent, CLI) now use a non-privileged user (`woodpecker`) by default.
```

Comment:
Should mention also uid/gid and that when upgrading need to change ownership in volumes/directory bindings

Reply:
But we dont set a fixed UID/GID right now or am I wrong? This should be fixed then in the rootless PR.